packages/read-model/src/Product/Detail/DetailProductViewFactory.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Product\Detail;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Utils\Utils;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade;
+use Shopsys\ReadModelBundle\Brand\BrandView;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionView;
+
+class DetailProductViewFactory
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade
+     */
+    protected $seoSettingFacade;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade $seoSettingFacade
+     */
+    public function __construct(
+        Domain $domain,
+        SeoSettingFacade $seoSettingFacade
+    ) {
+        $this->domain = $domain;
+        $this->seoSettingFacade = $seoSettingFacade;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice $sellingPrice
+     * @param int $mainCategoryId
+     * @param \Shopsys\ReadModelBundle\Image\ImageView[] $galleryImageViews
+     * @param \Shopsys\ReadModelBundle\Brand\BrandView|null $brandView
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionView $productActionView
+     * @param \Shopsys\ReadModelBundle\Parameter\ParameterView[] $parameterViews
+     * @return \Shopsys\ReadModelBundle\Product\Detail\DetailProductView
+     */
+    public function createFromProduct(
+        Product $product,
+        ProductPrice $sellingPrice,
+        int $mainCategoryId,
+        array $galleryImageViews,
+        ?BrandView $brandView,
+        ProductActionView $productActionView,
+        array $parameterViews
+    ): DetailProductView {
+        $domainId = $this->domain->getId();
```

Comment:
I thought we agreed to pass `domainId` as argument to make it more decoupled. Has anything changed or just missed one?

Reply:
It's not possible to decouple methods as under the hood is used many methods dependent internally on the current domain (`ProductCachedAttributesFacade::getProductSellingPrice`, ,ProductOnCurrentDomainElasticFacade::getVisibleProductById`, ...)

Reply:
That is so sad 😞